activestorage/lib/active_storage/attached/many.rb

```diff
@@ -28,12 +28,14 @@ def blobs
     #   document.images.attach(io: File.open("/path/to/racecar.jpg"), filename: "racecar.jpg", content_type: "image/jpeg")
     #   document.images.attach([ first_blob, second_blob ])
     def attach(*attachables)
-      if record.persisted? && !record.changed?
-        record.public_send("#{name}=", blobs + attachables.flatten)
-        record.save
-      else
-        record.public_send("#{name}=", (change&.attachables || blobs) + attachables.flatten)
+      if Array(attachables).any?
+        record.attachment_changes[name] = ActiveStorage::Attached::Changes::CreateMany.new(name,
```

Comment:
Format: I'd try to remove the long tabulation before each argument. I [think this format](https://github.com/rails/rails/blob/40d63cee31a99e57c05d86cab12507b535af135f/actionmailbox/test/unit/mail_ext/addresses_test.rb#L8-L13) is more idiomatic in the Rails codebase. Maybe one single line would do it here, I think the line can be shortened if you move it to the place suggested in my other comment.